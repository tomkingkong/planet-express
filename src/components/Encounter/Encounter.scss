.Encounter {
  cursor: pointer;
  padding: 1rem;
  height: 1000px;
}

.encounter__img {
  opacity: 0;
  border: 0.4rem outset rgba(224, 246, 255, 0.988);
  border-radius: 1rem;
  box-shadow: inset 0rem  0rem 1rem 0.4rem rgba(255, 255, 255, 0.386);
  background-size: cover;
  margin: 0rem auto;
  z-index: 999;
  transform: scale(0.2);
  transition: all 1s linear;
}

.encounter__title {
  color: white;
  background: rgba(0, 0, 0, 0.297);
  box-shadow: 0rem 0rem 1rem 0.5rem rgba(105, 120, 128, 0.365);
  border-radius: 0.25rem;
  font-size: 90%;
  text-align: center;
  overflow: scroll;
  padding: 0.8rem;
  margin: 0.5rem 0 0 0;
}

.encounter__desc {
  color: white;
  background: rgba(0, 0, 0, 0.297);
  font-size: 90%;
  line-height: 30px;
  text-align: justify;
  overflow: scroll;
  padding: 1rem 0;
  height: calc(40vh - 2.5rem);
  margin: 0.8rem 0 0 0;
}

.continue__btn {
  color: white;
  cursor: pointer;
  background: rgba(174, 180, 182, 0.639);
  border: none;
  box-shadow: 0rem 0rem 0.9rem 0.01rem rgb(218, 215, 215);
  font-size: 100%;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  padding: 0.5rem;
}
.continue__btn:hover {
  color: rgba(214, 219, 223, 0.639);
  background: rgba(86, 89, 91, 0.639);
  box-shadow: 0rem 0rem 0.9rem 0.01rem rgb(86, 89, 91);
}