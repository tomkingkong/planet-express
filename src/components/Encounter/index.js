import React, { Component } from 'react';
import { string, array } from 'prop-types';

import './Encounter.css';

export default class Encounter extends Component {
  constructor() {
    super();
    this.state = {
      showInfo: false
  }
  }
  return (
    <article className="Encounter">
      <div className="encounter__img" style={encounterImgStyle} />
      <h4 className="encounter__title">{name}</h4>
      <p className="encounter__desc">{description}</p>
      <button className="continue__btn" onClick={nextEvent}>continue</button>
    </article>
  )
}

const { string, array } = PropTypes;
Encounter.propTypes = {
  name: string,
  description: string,
  images_files: array
}