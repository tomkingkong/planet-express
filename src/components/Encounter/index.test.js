import React from 'react';
import { shallow } from 'enzyme';

import Encounter from '.';

describe('Encounter component', () => {
  let wrapper;
  let mockFiles;

  beforeEach(() => {
    mockFiles = [{file_url:'images'}, {file_url:'images'}, {file_url:'images'}]
    wrapper = shallow(
    <Encounter
      name={'tim'}
      description={'haha'}
      image_files={mockFiles}
    />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
