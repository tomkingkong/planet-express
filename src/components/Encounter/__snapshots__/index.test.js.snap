// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encounter component should match snapshot 1`] = `
<article
  className="Encounter"
>
  <div
    className="encounter__img"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(images)",
      }
    }
  />
</article>
`;

exports[`Encounter component should match snapshot with showInfo true 1`] = `
<article
  className="Encounter"
>
  <div
    className="encounter__img"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(images)",
      }
    }
  />
  <div
    className="encounter__info"
  >
    <h4
      className="encounter__title"
    >
      tim
    </h4>
    <p
      className="encounter__desc"
    >
      haha
    </p>
  </div>
</article>
`;
